### DISCUSSION TITLE
Vercel Preview Github Action

### DISCUSSION THREAD

**Discussion by jgoux** [vercel/vercel#8328] [Show and tell]
Hello all 👋,

I'm a software engineer at [Snaplet](https://snaplet.dev). We are working hard to enable developers across the world to don't have to worry about seeding or replicating their production databases into their dev or preview environments. 🙇‍♂️

Vercel is an integral part of this story, this week I investigated how we can integrate Snaplet with Vercel and I came up with this new GitHub Action: https://github.com/marketplace/actions/vercel-preview-deployments

My goal was to take control of **when** a deployment happens so I can orchestrate external resources creation before deploying a Vercel preview. It could be done with `vercel deploy --prebuilt` but the downside of using it is that you don't have the sweet feedback from the Vercel GitHub App anymore.

What I did instead is leveraging [deploy hooks](https://vercel.com/docs/concepts/git/deploy-hooks) so it's still possible to benefit from the Vercel GitHub App, while also being able to deploy when I decide, the best of both worlds! 😄

The action also has a `delete` option to clean up everything that could be produced on Vercel (mainly new environment variables), so it fits really well with a Pull Request lifecycle.

Don't hesitate to ping me if you want to know more, or if you want to try Snaplet as well (we are on [Discord](https://app.snaplet.dev/chat)). 🐱

**Comment by david0673** [comment]
this is awesome! any suggested workarounds for the limit of 5 deploy hooks? 

Separately, if I'm not mistaken, the action has moved to 
https://github.com/snaplet/vercel-action

**Reply by jgoux** [reply]
@david0673 I wasn't aware there was a limit for deploy hooks, that's...annoying. 😅

I don't think there is a workaround, unfortunately!

@leerob Is there any reason to limit the number of deploy hooks one can create?

**Reply by david0673** [reply]
For context, this is where the limit is mentioned:
https://vercel.com/docs/concepts/deployments/deploy-hooks#limits

it is not listed on the [general limits overview page](https://vercel.com/docs/concepts/limits/overview)

**Reply by jgoux** [reply]
@david0673 Thanks to Vercel team contribution, we no longer rely on deploy hooks, we switched to the deployment API. 🎊
It's released under v3: https://github.com/snaplet/vercel-action/releases/tag/v3.0.0

**Reply by david0673** [reply]
Thanks for the update! I've been collaborating with them in the QA process :)

Was the issue with Vercel's GitHub PR checks getting out of sync as a result of these changes resolved?

**Reply by jgoux** [reply]
> Was the issue with Vercel's GitHub PR checks getting out of sync as a result of these changes resolved

I'm not aware of this issue. @ofhouse did all the work, maybe he knows?

**Reply by ofhouse** [reply]
> Was the issue with Vercel's GitHub PR checks getting out of sync as a result of these changes resolved?

Not resolved yet, was more important for us to get the update out first, so that you are no longer blocked by the Deploy Hook limit ✌️

That the PR check got out of sync is a result that we now using another API endpoint for that internally.
It's on our radar and we should be able to fix it on our side without any update needed.

Will update here (and through our customer support) once we have it resolved, cheers 🍻

**Reply by ofhouse** [reply]
Finally found some time to investigate further.
Problem is that I thought `GITHUB_SHA` would refer to the commit that is part of a PR (See this discussion: https://github.com/orgs/community/discussions/26325) 🙈 

So the check is added from our side, but unfortunately to a ghost commit that never shows up in the PR.
Made a PR here to fix it: https://github.com/snaplet/vercel-action/pull/10

**Reply by jgoux** [reply]
Thanks @ofhouse for your contribution, I released it under `v3`/`v3.0.1` 👍

**Reply by ofhouse** [reply]
Great @jgoux, thanks for your super fast response, really appreciate 👍
